### ISSUE TITLE
Bridge difflow flowsheets to pounce and discopt

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/differentiable-flowsheets#203]
## Summary

`difflow` has no connection to `pounce` or `discopt`:

```
$ grep -rl "pounce\|discopt" src/ --include=*.py | wc -l
0
```

The only solver-facing export anywhere is `planning/lp.py::to_pyomo()`, which emits the linearized delta-base LP, not the flowsheet. So every capability those two packages provide has to be reached by hand-rolling the model a second time.

That is a larger loss than it looks, because both already accept exactly the kind of object difflow produces.

## What the other side offers

`pounce.jax.from_jax(f, g)` builds a solver `Problem` straight from JAX-traced objective and constraint functions, deriving the gradient, the constraint Jacobian with sparsity detected from a one-shot dense probe, and the Lagrangian Hessian via `jax.grad`/`jacrev`/`hessian`, then JIT-compiling them. `pounce.jax.solve` wraps the solve in `jax.custom_vjp`, with the backward pass applying the implicit function theorem to the KKT system, so an outer JAX computation can differentiate through the optimization. `JaxProblem` supports build-once, solve-many with factor reuse.

`discopt.modeling.implicit(g, ...)` consumes a residual `g(u, v) = 0` and compiles it to a differentiable inner Newton solve whose derivatives come from `jax.lax.custom_root`, which supports higher-order AD, so a second-order NLP solver's Hessian works through it.

Both are close to what difflow already computes. Neither is reachable today.

## Proposed implementation

Two adapters, small, in a new `difflow/solvers/` module.

**1. A flat NLP view of a flowsheet.** Given a flowsheet, a set of decision variables and a set of specifications, return

```python
def as_nlp(flowsheet, decisions, specs) -> tuple[Callable, Callable, Bounds]
```

where `f(x)` is the objective and `g(x)` the constraint body, both JAX-traceable. That is exactly `pounce.jax.from_jax`'s input contract, so `pounce` support is then a two-line convenience wrapper, and `pounce.jax.solve` makes the whole design problem differentiable end to end.

**2. A residual view.** Given a unit or a section, return `g(u, v)` with `v` the internal states and `u` the inputs and parameters, suitable for `discopt.modeling.implicit`. `eo_solver.py` already builds residuals internally; this exposes them.

## The constraint that shapes the design

Worth recording in the docs alongside the adapter, because it is easy to get wrong: `discopt`'s `CustomCall`, which is what `dm.custom` and therefore `dm.implicit` produce, is restricted to the **local NLP path only**, carries no global optimality certificate, makes relaxation compilation and `.nl` export raise, and **the solver raises if integer or binary variables are present**, since spatial branch and bound has no valid node relaxation for an opaque callable.

So a difflow flowsheet cannot be wrapped as a UDF and then have binaries put around it in one discopt model. The adapter should say so plainly, so users reach for decomposition rather than discovering the restriction at solve time. Where integrality and global bounds are actually wanted, the model has to be written in discopt's algebraic language instead.

## Why it is worth doing

`pounce` post-optimal sensitivity is a sIPOPT port and gives the reduced Hessian, which is what constraint back-off and advanced-step NMPC both need. `pounce.trf_minimize` implements the trust-region filter for mixed glass-box and black-box models, which is the multi-fidelity manager for a model with several levels of physical detail. `discopt` brings GDP with logic-based outer approximation, spatial branch and bound, bilevel and MPEC, robust and stochastic formulations, multi-objective scalarization, DAE collocation, parameter estimation with Fisher information, and neural-network embedding.

None of that needs reimplementing in difflow. It needs a bridge.

## References

Kitchin, POUNCE, doi:10.5281/zenodo.20387011. Kitchin, discopt, doi:10.5281/zenodo.19762815.
